pydoc/developer/BENCHMARK.md

```diff
@@ -0,0 +1,470 @@
+
+The PyTest benchmark structure allows the flexibility of running the entire suite, a single file, 
+or a single benchmark.
+
+## Running The Full Suite
+In most cases, running the full benchmark suite is desired. The simplest way to do this is to navigate into the 
+root-level of arkouda and run `make benchmark`
+
+This will run the entire benchmark suite with the following command:
+```commandline
+python3 -m pytest -c benchmark.ini --benchmark-autosave --benchmark-storage=file://benchmark_v2/.benchmarks
+```
+
+## Running Single Files or Tests
+
+In instances where a single test or set of tests needs to be ran, use the `-k <name_of_file>.py` argument. For these 
+examples, `encoding_benchmark.py` and the specific benchmark from within the file `bench_encode` is used.
+
+```commandline
+python3 -m pytest -c benchmark.ini --benchmark-autosave --benchmark-storage=file://benchmark_v2/.benchmarks -k encoding_benchmark.py
+```
+
+Running this command, you can expect to see an output table similar to this
+```commandline
+benchmark_v2/encoding_benchmark.py ....                                                                                                                                       [100%]
+Saved benchmark data in: <Arkouda_root>/benchmark_v2/.benchmarks/Linux-CPython-3.9-64bit/0014_31de39be8b19c76d073a8999def6673a305c250d_20230405_145759_uncommited-changes.json
+
+-------------------------------------------------------------------- benchmark 'Strings_EncodeDecode': 4 tests ---------------------------------------------------------------------
+Name (time in ms)          Min               Max              Mean            StdDev            Median               IQR            Outliers       OPS            Rounds  Iterations
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bench_encode[idna]      3.3304 (1.0)      9.2561 (2.10)     4.7544 (1.27)     2.5306 (6.18)     3.8075 (1.11)     1.9012 (3.62)          1;1  210.3306 (0.79)          5           1
+bench_encode[ascii]     3.3805 (1.02)     4.8800 (1.10)     3.7336 (1.0)      0.6465 (1.58)     3.4231 (1.0)      0.5246 (1.0)           1;1  267.8380 (1.0)           5           1
+bench_decode[idna]      3.4444 (1.03)     4.4177 (1.0)      3.7852 (1.01)     0.4097 (1.0)      3.5622 (1.04)     0.5837 (1.11)          1;0  264.1882 (0.99)          5           1
+bench_decode[ascii]     3.4621 (1.04)     4.9177 (1.11)     4.2250 (1.13)     0.6125 (1.50)     4.0197 (1.17)     0.9991 (1.90)          2;0  236.6864 (0.88)          5           1
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+```
+Similarly, to only run a single test within a file, specify the test name with the `-k` flag instead of a filename

```

Comment:
```suggestion
Similarly, to only run a single test within a file, specify the test name with the `-k` flag instead of a filename. The following example will run only the `bench_encode` benchmark.
```